loklak/loklak_server
# Installation page

**@ Codacy fail:**
Codacy is failing on this PR as libs like jquery get reviewed, too, which do not pass the test. The code contributions from me are fine.

This is a working version of the installation page https://github.com/loklak/loklak_server/issues/519
Warning: requires bash for shell interaction!

It currently offers:
- admin account creation
- a user registration setting
- general settings like host- and shortlink url, peername, backend settings
- smtp settings (with test option)
- https settings

The installation is started with an own start script, in it's own main class. Launching it will be offered by the normal start script.
In the process I also made some changes/additions to the email handler

How to test:
- clonegit@github.com:treba123/loklak_server.git
- checkout development
- build and start via start.sh
- choose to do the installation
- go through the dialog
- afterwards, the normal server should be started

**Codacy:**
I adopted most recommendations, most remaining issues are with external files which I have little influence on.
